corehq/apps/api/resources/v0_5.py

```diff
@@ -268,15 +268,14 @@ def obj_create(self, bundle, **kwargs):
             bundle.obj.save()
         except Exception:
             if bundle.obj._id:
-                bundle.obj.retire(deleted_by=bundle.request.user, deleted_via=USER_CHANGE_VIA_API)
+                bundle.obj.retire(bundle.request.domain, deleted_by=bundle.request.couch_user,
```

Comment:
how come this is `retire` and the other one (https://github.com/dimagi/commcare-hq/pull/29901/files#diff-efc09ad10beb9e39276a54e59070e11ec9947779bc6a4413196f0a086e70c841R376) is `delete`?

Reply:
Great catch.

Tracked this down to https://github.com/dimagi/commcare-hq/pull/3715/files
(was hard because the same line was edited between its addition in https://github.com/dimagi/commcare-hq/commit/f821f5f0cd5ddb76e8bc745a049427e9725e4718 and last history edit in https://github.com/dimagi/commcare-hq/pull/9733/commits/8baacfbdd556f15909df4971367fee90c453cb3c)

Reply:
thanks for digging that up - it seems the original issue is that ES wasn't removing 'deleted' users so we used 'retire' instead. I wonder if that's still an issue? (fine to leave as is for this PR)

Reply:
Looks like its worth reconsidering. Not sure i am connecting this correctly, I believe if a user is created with a username and then retired/deleted immediately, the username gets blocked and can't be reused. If that is true, its surprising that that has not come as the issue till now.